plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/LaunchAppEngineStandardHandlerTest.java

```diff
@@ -0,0 +1,147 @@
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.internal.ModuleType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * Tests for the {@link LaunchAppEngineStandardHandler}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class LaunchAppEngineStandardHandlerTest {
+  private static final IProjectFacetVersion APPENGINE_STANDARD_FACET_VERSION_1 =
+      ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID).getVersion("1");
+
+  @Rule
+  public ServerTracker tracker = new ServerTracker();
+
+  private LaunchAppEngineStandardHandler handler;
+
+  @Rule
+  public TestProjectCreator appEngineStandardProject1 =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+          WebFacetUtils.WEB_25, APPENGINE_STANDARD_FACET_VERSION_1));
+  @Rule
+  public TestProjectCreator appEngineStandardProject2 =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+          WebFacetUtils.WEB_25, APPENGINE_STANDARD_FACET_VERSION_1));
+
+  private IModule module1;
+  private IModule module2;
+
+  @Before
+  public void setUp() throws CoreException {
+    handler = new LaunchAppEngineStandardHandler();
+    handler.mockLaunch = true;
+  }
+
+  /** Must provide a real project as validators fail. */
+  private static IModule mockAppEngineStandardModule(String serviceId, IProject project)
+      throws CoreException {
+    ModuleType webModuleType = new ModuleType("jst.web", "2.5");
+    IModule module = mock(IModule.class);
+
+    when(module.getName()).thenReturn(serviceId);
+    when(module.getModuleType()).thenReturn(webModuleType);
+    when(module.getProject()).thenReturn(project);
+
+    IFolder webinf = WebProjectUtil.getWebInfDirectory(project);
+    IFile descriptorFile = webinf.getFile("appengine-web.xml");
+    assertTrue(descriptorFile.exists());
+    descriptorFile.setContents(appEngineForService(serviceId), IFile.FORCE, null);
+
+    return module;
+  }
+
+  private static InputStream appEngineForService(String serviceId) {
+    StringBuilder contents = new StringBuilder();
+    contents.append("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n");
+    contents.append("<appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\">\n");
+    contents.append("<service>").append(serviceId).append("</service>\n");
+    contents.append("</appengine-web-app>\n");
+    return new ByteArrayInputStream(contents.toString().getBytes(StandardCharsets.UTF_8));
+  }
+
+  @Test
+  public void testWithOneModule() throws ExecutionException, CoreException {
+    module1 = mockAppEngineStandardModule("default", appEngineStandardProject1.getProject());
```

Comment:
If you're always initializing module1 and module2 in test methods, you might as well make them local variables.